ecs-agent/wsclient/client_test.go

```diff
@@ -412,13 +432,55 @@ func TestCtxCancel(t *testing.T) {
 
 	types := []interface{}{ecsacs.AckRequest{}}
 	cs := getTestClientServer(mockServer.URL, types, 2)
-	require.NoError(t, cs.Connect())
+	timer, err := cs.Connect(mockDisconnectTimeoutMetricName, DisconnectTimeout, DisconnectJitterMax)
+	require.NoError(t, err)
+	defer timer.Stop()
+
 	assert.True(t, cs.IsReady(), "expected websocket connection to be ready")
 
 	go func() {
 		messageError <- cs.ConsumeMessages(ctx)
 	}()
 	// Cancel the context.
 	cancel()
-	assert.EqualError(t, <-messageError, "context canceled")
+	err = <-messageError
+	assert.Equal(t, err.Error(), context.Canceled.Error(), "Context canceled error expected.")
+}
+
+func TestPeriodicDisconnect(t *testing.T) {
+	closeWS := make(chan []byte)
+	defer close(closeWS)
+
+	mockServer, _, _, errChan, _ := utils.GetMockServer(closeWS)
+	mockServer.StartTLS()
+
+	messageError := make(chan error)
+	ctx := context.Background()
+	types := []interface{}{ecsacs.AckRequest{}}
+
+	// Setting a higher rwtimeout to allow disconnect due to periodic timeouts.
+	cs := getTestClientServer(mockServer.URL, types, 20)
+	// Setting up a lower disconnect timer value for testing.
+	disconnectTimeout := 10 * time.Second
+	disconnectTimeoutJitter := 2 * time.Second
+	timer, err := cs.Connect(mockDisconnectTimeoutMetricName, disconnectTimeout, disconnectTimeoutJitter)
+	require.NoError(t, err)
+	defer timer.Stop()
+	assert.True(t, cs.IsReady(), "expected websocket connection to be ready")
+
+	// Using time difference to ensure that the disconnect was
+	go func() {
+		startTime := time.Now()
+		messageError <- cs.ConsumeMessages(ctx)
+		endTime := time.Now()
+		timeDiff := endTime.Sub(startTime)
+		assert.True(t, timeDiff > disconnectTimeout,
+			"ConsumeMessages should be alive for more than disconnectTimeout.")
```

Comment:
I'm not sure that this statement is accurate. ConsumeMessages can be alive for the amount captured in disconnectTimeout, or potentially more (as much as disconnectTimeoutJitter longer). Perhaps a more accurate statement would be something like "ConsumeMessages should be not be closed before disconnectTimeout has elapsed"